src/configuration/configuration.ts

```diff
@@ -0,0 +1,84 @@
+"use strict";
+
+import * as vscode from 'vscode';
+
+export type OptionValue = number | string | boolean;
+
+/**
+ * Every Vim option we support should
+ * 1. Be added to contribution section of `package.json`.
+ * 2. Named as `vim.{optionName}`, `optionName` is the name we use in Vim.
+ * 3. Define a public property in `Configuration `with the same name and a default value.
+ *    Or define a private propery and define customized Getter/Setter accessors for it.
+ *    Always remember to decorate Getter accessor as @enumerable()
+ * 4. If user doesn't set the option explicitly
+ *    a. we don't have a similar setting in Code, initialize the option as default value.
+ *    b. we have a similar setting in Code, use Code's setting.
+ *
+ * Vim option override sequence.
+ * 1. `:set {option}` on the fly
+ * 2. TODO .vimrc.
+ * 2. `vim.{option}`
+ * 3. VS Code configuration
+ * 4. VSCodeVim flavored Vim option default values
+ *
+ */
+export class Configuration {
+  private static _instance: Configuration | null;
+
+  constructor() {
+    /**
+     * Load Vim options from User Settings.
+     */
+    let vimOptions = vscode.workspace.getConfiguration("vim");
+    /* tslint:disable:forin */
+    // Disable forin rule here as we make accessors enumerable.`
+    for (const option in this) {
+      const vimOptionValue = vimOptions[option];
+      if (vimOptionValue !== null && vimOptionValue !== undefined) {
+        this[option] = vimOptionValue;
+      }
+    }
+  }
+
+  public static getInstance(): Configuration {
+    if (Configuration._instance == null) {
```

Comment:
Nevermind, we read use this A LOT in our code. The alternative I had in mind initially was to read the configs from the file every time.